Protos/step_by_step.proto

```diff
@@ -19,16 +22,30 @@ service StepByStepService {
 
     // Checks whether a instruction is implemented correctly.
     rpc CheckInstruction (CheckInstructionRequest) returns (CheckInstructionResponse);
+
+    // Sets the filepath of every instruction to the newly created .cs file.
+    rpc SetNewFilePath (SetFilePathRequest) returns (SetFilePathResponse);
 }
 
 // GetInstructionSets request message.
-message GetInstructionSetsRequest {
-}
+message GetInstructionSetsRequest { }
 
 // GetInstructionSets response message.
 message GetInstructionSetsResponse {
-    // List of instruction sets which are available for step-by-step mode.
-    repeated InstructionSet instruction_sets = 1;
+    // InstructionSets that the InstructionsCreator can make.
+    repeated patternpal.Recognizer recognizers = 1;
+}
+
+// GetInstructionSet request message.
+message GetInstructionSetRequest {
+    // Name of the PatternPal
+    patternpal.Recognizer selected_combobox = 1;
```

Comment:
You're mixing concerns here, why should this message know anything about how the instruction set was selected?

Reply:
I have moved everything to do with getting the instruction set to the StepByStepInstructionsViewmodel. Instead of the StepByStepListViewModel getting the instruction set based on what was selected in the combox it instead it provides the constructor of the StepByStepInstructionsViewModel with the recognizer that was selected and uses that recognizer to return the instruction from the service.

Reply:
I have additionally removed the static InstructionSetHolder, almost everything in the service now requires a recognizer to retrieve the instruction. It is now "stateless".